modules/acm/src/main/java/edu/gemini/epics/acm/CaCommandMonitorImpl.java

```diff
@@ -81,6 +81,43 @@ public void waitDone() throws InterruptedException {
         }
     }
 
+    private boolean isActive() {
+        return currentState == State.BUSY;
+    }
+
+    @Override
+    public State waitInactive(long timeout, TimeUnit unit)
+            throws TimeoutException, InterruptedException {
+        lock.lock();
+        State state = currentState;
+        try {
+            while (isActive()) {
```

Comment:
This loop is  a bit worrisome. Could we ever be stuck in an infinite loop?

Reply:
No, because eventually the timeout will be triggered.

Reply:
On second thought, there is a small issue here, in that if after x seconds the condition is triggered spuriously (the reason of the loop), the `await` will be  called again with the original timeout value, making effectively equal to timeout + x.
I'll put a Jira ticket to fix it.